src/notification-hub/azext_notification_hub/_params.py

```diff
@@ -0,0 +1,190 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from argcomplete.completers import FilesCompleter
+from knack.arguments import CLIArgumentType
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    get_location_type,
+    file_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+
+from ._validators import validate_cert_file
+
+
+def load_arguments(self, _):
+    payload_type = CLIArgumentType(
+        options_list=['--payload'],
+        help='The payload for the message in JSON format. '
+    )
+
+    with self.argument_context('notification-hub namespace create') as c:
+        c.argument('namespace_name', options_list=['--name', '-n'], help='The namespace name.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
+        c.argument('tags', tags_type)
+        c.argument('sku_name', arg_type=get_enum_type(['Free', 'Basic', 'Standard']), options_list=['--sku'], help='Name of the notification hub sku')
+
+    with self.argument_context('notification-hub namespace update') as c:
+        c.argument('namespace_name', options_list=['--name', '-n'], help='The namespace name.')
+        c.argument('tags', tags_type)
+        c.argument('sku_name', arg_type=get_enum_type(['Free', 'Basic', 'Standard']), options_list=['--sku'], help='Name of the notification hub sku')
+
+    with self.argument_context('notification-hub namespace delete') as c:
+        c.argument('namespace_name', id_part="name", options_list=['--name', '-n'], help='The namespace name.')
+
+    with self.argument_context('notification-hub namespace show') as c:
```

Comment:
actually we can put `c.argument('namespace_name', id_part="name", options_list=['--name', '-n'], help='The namespace name.')` under the context `self.argument_context('notification-hub namespace')`. However, since this is auto generated. Currently, it's fine. But we can discuss this later together.

Reply:
maybe the limitation of code generation.